src/strands/experimental/multiagent_session/multiagent_events.py

```diff
@@ -0,0 +1,87 @@
+"""Multi-agent execution lifecycle events for hook system integration.
+
+This module defines event classes that are triggered at key points during
+multi-agent orchestrator execution, enabling hooks to respond to lifecycle
+events for purposes like persistence, monitoring, and debugging.
+
+Event Types:
+- Initialization: When orchestrator starts up
+- Before/After Graph: Start/end of overall execution
+- Before/After Node: Start/end of individual node execution
+"""
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from ...hooks.registry import HookEvent
+from .multiagent_state import MultiAgentState
+
+if TYPE_CHECKING:
+    from ...multiagent.base import MultiAgentBase
+
+
+@dataclass
+class MultiAgentInitializationEvent(HookEvent):
```

Comment:
should we also do a `MultiAgentHookEvent` as a generic event? It would allow people to create generic hook implementations and vend them. The end users can use it in whichever event they want